src/app/components/normal-component/chat/chat.component.ts

```diff
@@ -1,85 +1,131 @@
-    public chat: IMessage[];
-    public userMessage: string;
+    public connected: boolean = false;
+    public chat: IMessage[] = [];
     public readMessages: number = 0;
-    public unreadMessages: number = 0;
+    public unreadMessagesText: string = '';
 
     public visible: boolean = true;
 
+    @ViewChild('chatDiv') chatDiv: ElementRef<HTMLElement>;
+
     constructor(private chatService: ChatService,
                 private authenticationService: AuthenticationService) {
         display(ChatComponent.VERBOSE, 'ChatComponent constructor');
     }
     public ngOnInit(): void {
         display(ChatComponent.VERBOSE, 'ChatComponent.ngOnInit');
 
-        if (this.chatId == null || this.chatId === '') throw new Error('No chat to join mentionned');
+        assert(this.chatId != null && this.chatId !== '', 'No chat to join mentionned');
 
         this.authenticationService.getJoueurObs()
             .subscribe((joueur: AuthUser) => {
                 if (this.isConnectedUser(joueur)) {
                     display(ChatComponent.VERBOSE, JSON.stringify(joueur) + ' just connected');
-                    this.userName = joueur.pseudo;
+                    this.userName = MGPOptional.of(joueur.pseudo);
+                    this.connected = true;
                     this.loadChatContent();
                 } else {
                     display(ChatComponent.VERBOSE, 'No User Logged');
-                    this.showDisconnectedChat();
+                    this.userName = MGPOptional.empty();
+                    this.connected = false;
                 }
             });
     }
+    public ngAfterViewChecked(): void {
+        this.scrollToBottomIfNeeded();
+    }
     public isConnectedUser(joueur: { pseudo: string; verified: boolean;}): boolean {
-        return joueur && joueur.pseudo &&
-               joueur.pseudo !== '' &&
-               joueur.pseudo !== 'null' &&
-               joueur.pseudo !== 'undefined';
+        return joueur && joueur.pseudo && joueur.pseudo !== '';
     }
     public loadChatContent(): void {
         display(ChatComponent.VERBOSE, 'User \'' + this.userName + '\' logged, loading chat content');
 
-        this.chatService.startObserving(this.chatId, (id: IChatId) => this.updateMessages(id));
+        this.chatService.startObserving(this.chatId, (id: IChatId) => {
+            this.updateMessages(id);
+        });
     }
     public updateMessages(iChatId: IChatId): void {
         this.chat = iChatId.doc.messages;
         const nbMessages: number = this.chat.length;
-        if (this.visible === false) {
-            this.unreadMessages = nbMessages - this.readMessages;
+        if (this.visible === false || this.isNearBottom === false) {
+            this.updateUnreadMessagesText(nbMessages - this.readMessages);
         } else {
             this.readMessages = nbMessages;
-            this.unreadMessages = 0;
+            this.updateUnreadMessagesText(0);
+            this.scrollToBottom();
+        }
```

Comment:
prefer
if ( A and B) SECOND else FIRST
to
if (!A or !B) FIRST else SECOND :D !